Spec/RealtimeClientPresence.swift

```diff
@@ -2889,76 +2889,81 @@ class RealtimeClientPresence: QuickSpec {
             }
 
             // RTP15e
-            let cases: [String:(ARTRealtimePresence, Optional<(ARTErrorInfo?)->Void>)->()] = [
-                "enterClient": { $0.enterClient("john", data: nil, callback: $1) },
-                "updateClient": { $0.updateClient("john", data: nil, callback: $1) },
-                "leaveClient": { $0.leaveClient("john", data: nil, callback: $1) }
-            ]
-            for (testCase, performMethod) in cases {
-                context(testCase) {
-                    it("should implicitly attach the Channel") {
-                        let client = ARTRealtime(options: AblyTests.commonAppSetup())
-                        defer { client.dispose(); client.close() }
-                        let channel = client.channels.get("test")
-
-                        expect(channel.state).to(equal(ARTRealtimeChannelState.initialized))
-                        waitUntil(timeout: testTimeout) { done in
-                            //Call: enterClient, updateClient and leaveClient
-                            performMethod(channel.presence) { errorInfo in
-                                expect(errorInfo).to(beNil())
-                                done()
-                            }
-                            expect(channel.state).to(equal(ARTRealtimeChannelState.attaching))
+            func reusableTestsTestPresencePerformMethod(_ performMethod: @escaping (ARTRealtimePresence, Optional<(ARTErrorInfo?)->Void>)->()) {
+                it("should implicitly attach the Channel") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.dispose(); client.close() }
```

Comment:
I don't like these inline calls and the use of the `;`. It looks better in new lines.
```suggestion
                    defer { 
                        client.dispose()
                        client.close() 
                    }
```

Reply:
I agree with all your comments about the formatting here. However, I didn't want to touch the formatting of the existing code, to make it clear that all I've done is move the existing code into a new function.

I'm planning to run a code formatter over all of the test code once the migration is complete, anyway, so this will get fixed then. Are you happy with this?